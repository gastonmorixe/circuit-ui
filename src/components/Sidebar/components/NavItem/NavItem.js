import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'react-emotion';

const baseStyles = ({ theme }) => css`
  display: flex;
  align-items: center;
  justify-content: flex-start;
  height: auto;
  margin: 16px;
  padding: 4px;
  cursor: pointer;
  color: ${theme.colors.n500};
`;

const hoverStyles = ({ theme, selected }) => css`
  @media (hover: hover) {
    &:hover {
      color: ${!selected && theme.colors.n300};
    }
  }
`;

const activeStyles = ({ theme, selected }) => css`
  color: ${selected && theme.colors.n100};
  font-weight: ${selected && 'bold'};
  &:active {
    color: ${theme.colors.n100};
    font-weight: bold;
  }
`;

const iconWrapperStyles = () => css`
  margin-right: 12px;
`;

const ListItem = styled('li')(baseStyles, hoverStyles, activeStyles);
const IconWrapper = styled('div')(iconWrapperStyles);

const NavItem = ({ children, icon, selected, onClick }) => (
  <ListItem selected={selected} onClick={onClick}>
    {icon && <IconWrapper>{icon}</IconWrapper>}
    {children}
  </ListItem>
);

NavItem.propTypes = {
  /**
   * The children component passed to the Sidebar
   */
  children: PropTypes.node,
  /**
   * The icon to be shown befor the Item title
   */
  icon: PropTypes.node,
  /**
   * Tells if the item is selected
   */
  selected: PropTypes.bool,
  /**
   * The onClick method to handle the click event on NavItems
   */
  onClick: PropTypes.func
};

export default NavItem;
