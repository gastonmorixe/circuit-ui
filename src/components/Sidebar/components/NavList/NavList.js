import PropTypes from 'prop-types';
import styled, { css } from 'react-emotion';

const baseStyles = () => css`
  height: auto;
  justify-self: flex-start;
  overflow-y: auto;
  width: 100%;
`;

const NavList = styled('ul')(baseStyles);

NavList.propTypes = {
  /**
   * The children component passed to the Sidebar
   */
  children: PropTypes.node
};

export default NavList;
