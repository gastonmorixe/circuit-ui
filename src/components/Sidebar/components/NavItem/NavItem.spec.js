import React from 'react';

import NavList from '../NavList';
import NavItem from './NavItem';

describe('NavItem', () => {
  describe('styles', () => {
    it('should render with default styles and match the snapshot', () => {
      const props = {
        selected: false,
        onClick: jest.fn()
      };
      const actual = create(<NavItem {...props} />);
      expect(actual).toMatchSnapshot();
    });

    it('should render with selected state styles and match the snapshot', () => {
      const props = {
        selected: true,
        onClick: jest.fn()
      };
      const actual = create(<NavItem {...props} />);
      expect(actual).toMatchSnapshot();
    });

    it('should render children', () => {
      const wrapper = shallow(
        <NavItem>
          <span data-selector="child">text node</span>
        </NavItem>
      );
      const actual = wrapper.find('[data-selector="child"]');

      expect(actual).toHaveLength(1);
      expect(actual.text()).toEqual('text node');
    });

    it('should render an icon', () => {
      const wrapper = shallow(
        <NavItem icon={<div />}>
          <span data-selector="child">text node</span>
        </NavItem>
      );
      const actual = wrapper.find('IconWrapper');

      expect(actual).toHaveLength(1);
    });
  });

  describe('accessibility', () => {
    it('should meet accessibility guidelines', async () => {
      const wrapper = renderToHtml(
        <NavList>
          <NavItem />
        </NavList>
      );
      const actual = await axe(wrapper);
      expect(actual).toHaveNoViolations();
    });
  });
});
