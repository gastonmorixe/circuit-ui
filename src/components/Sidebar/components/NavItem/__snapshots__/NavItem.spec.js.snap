// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem styles should render with default styles and match the snapshot 1`] = `
.circuit-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  height: auto;
  margin: 16px;
  padding: 4px;
  cursor: pointer;
  color: #9DA7B1;
}

@media (hover:hover) {
  .circuit-0:hover {
    color: #D8DDE1;
  }
}

.circuit-0:active {
  color: #FAFBFC;
  font-weight: bold;
}

<li
  className="circuit-0 circuit-1"
  onClick={[MockFunction]}
  selected={false}
/>
`;

exports[`NavItem styles should render with selected state styles and match the snapshot 1`] = `
.circuit-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  height: auto;
  margin: 16px;
  padding: 4px;
  cursor: pointer;
  color: #9DA7B1;
  color: #FAFBFC;
  font-weight: bold;
}

@media (hover:hover) {

}

.circuit-0:active {
  color: #FAFBFC;
  font-weight: bold;
}

<li
  className="circuit-0 circuit-1"
  onClick={[MockFunction]}
  selected={true}
/>
`;
